spockk-compiler-plugin/src/main/kotlin/io/github/pshevche/spockk/compilation/transformer/mock/MockingApiTransformer.kt

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.pshevche.spockk.compilation.transformer.mock
+
+import io.github.pshevche.spockk.compilation.SpockkConstants
+import io.github.pshevche.spockk.compilation.common.BaseSpockkIrElementTransformer
+import io.github.pshevche.spockk.compilation.transformer.SpockkIrFactory
+import org.jetbrains.kotlin.ir.IrStatement
+import org.jetbrains.kotlin.ir.UNDEFINED_OFFSET
+import org.jetbrains.kotlin.ir.declarations.*
+import org.jetbrains.kotlin.ir.expressions.IrCall
+import org.jetbrains.kotlin.ir.expressions.IrConst
+import org.jetbrains.kotlin.ir.expressions.IrExpression
+import org.jetbrains.kotlin.ir.expressions.IrTypeOperatorCall
+import org.jetbrains.kotlin.ir.expressions.impl.IrClassReferenceImpl
+import org.jetbrains.kotlin.ir.expressions.impl.IrConstImpl
+import org.jetbrains.kotlin.ir.symbols.UnsafeDuringIrConstructionAPI
+import org.jetbrains.kotlin.ir.types.IrTypeSystemContext
+import org.jetbrains.kotlin.ir.types.IrTypeSystemContextImpl
+import org.jetbrains.kotlin.ir.types.classOrNull
+import org.jetbrains.kotlin.ir.util.findDeclaration
+import org.jetbrains.kotlin.ir.util.isSubtypeOf
+import org.jetbrains.kotlin.ir.util.toIrConst
+import org.jetbrains.kotlin.name.Name
+import java.util.stream.Collectors
+
+@OptIn(UnsafeDuringIrConstructionAPI::class)
+internal class MockingApiTransformer(irFactory: SpockkIrFactory, private val spec: IrClass) :
+  BaseSpockkIrElementTransformer() {
+
+  private val pluginContext = irFactory.pluginContext
+  private val irBuiltIns = pluginContext.symbols.irBuiltIns
+  private val specInternalsClass = pluginContext.referenceClass(SpockkConstants.SPEC_INTERNALS_CLASS_ID)!!
+  private val kClassJavaPropGetter =
+    pluginContext.referenceProperties(SpockkConstants.KCLASS_JAVA_PROPERTY_ID).first().owner.getter!!
+
+  companion object {
+
+    private val MOCK_METHODS: Map<Name, Name> = setOf(
+      "Mock",
+      "Stub",
+      "Spy"
+    ).stream()
+      .collect(
+        Collectors.toMap(
+          { f -> Name.identifier(f) },
+          { f -> Name.identifier(f + "Impl") })
+      )
+  }
+
+  fun rewriteMockingApi() {
+    spec.declarations.forEach { declaration ->
+      if (declaration is IrFunction || declaration is IrProperty) {
+        declaration.accept(this, null)
+      }
+    }
+  }
+
+  override fun visitVariable(declaration: IrVariable): IrStatement {
```

Comment:
❔ Could you please add a comment explaining what type of statements this applies to? I found comments similar to the ones in `WhereBlockRewriter#rewriteBinaryWhereStat` very helpful to understand what exactly we process in each block.